server/config/passport.ts

```diff
@@ -27,10 +27,19 @@ passportUse(new LocalStrategy({
         .select('username email hashed_password salt')
         .exec((err, user) => {
             if (err) return done(err);
+            let errorObject = { message: 'wrong username or password' };
             if (!user || !user.authenticate(password)) {
-                return done(null, false, { message: 'wrong username or password' });
+                return done(null, false, errorObject);
```

Comment:
Oh, yeah, I found this pattern particularly weird as well! It looks like black magic. :)